site/cds_rdm/resolver.py

```diff
@@ -8,16 +8,41 @@
 
 """Resolver."""
 
+from flask import g
 from invenio_pidstore.models import PersistentIdentifier
+from invenio_rdm_records.proxies import current_rdm_records_service
+from invenio_search.engine import dsl
 
 
 def get_pid_by_legacy_recid(legacy_recid):
     """Get record by pid value."""
+    # Get the object uuid from pidstore
     recid = PersistentIdentifier.query.filter_by(
         pid_value=legacy_recid, pid_type="lrecid"
     ).one()
-    obj_uuid = recid.object_uuid
-    pid = PersistentIdentifier.query.filter_by(
-        object_uuid=obj_uuid, pid_type="recid"
+
+    # Use the object uuid to get the parent pid value
+    parent_pid = PersistentIdentifier.query.filter_by(
+        object_uuid=recid.object_uuid, pid_type="recid"
     ).one()
-    return pid.pid_value
+
+    return parent_pid.pid_value
+
+
+def get_record_by_version(parent_pid_value, version):
+    """Get record by parent pid value and version."""
+    latest_record = current_rdm_records_service.read_latest(
+        identity=g.identity, id_=parent_pid_value
+    )
+    if not version or version == "all" or latest_record["versions"]["index"] == version:
+        return latest_record
+
+    # Use the version number to get the desired record pid value
+    record = current_rdm_records_service.search_versions(
+        identity=g.identity,
+        id_=latest_record["id"],
+        extra_filter=dsl.Q("term", **{"versions.index": version}),
+    ).to_dict()["hits"]["hits"]
+    if not record:
+        return latest_record
```

Comment:
Using [this function](https://github.com/inveniosoftware/invenio-drafts-resources/blob/a5bb49b5dc00830e050207d919ab14ea1a3a47a3/invenio_drafts_resources/services/records/service.py#L150)
If we raise and handle exception in the linked function, it wouldn't change how we handle the record not being found here. If we raise an exception there, and handle it here, then we'll also have to handle the exception everywhere this function is being used, also it will error out for the api. Or maybe I missed something here? Please let me know 😅

Reply:
ping @ntarocco how about having specific 404 page with a link pointing to lastest version ? in case of the specific requested one was not there (deleted). I agree with Javi that redirecting by default to the latest, when you requested a specific version in the URL, without any warning, might be confusing for the user.

Reply:
We discussed IRL with Saksham and decided to go with the latest version.
I agree that a 404 with a link to the latest might be better indeed.
However, it depends on how hard it is. @sakshamarora1 can you check? If it takes more than half a day, let's go with the current implementation.

Reply:
Added the template!